tests/e2e/encrypted.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+#
+# Copyright Confidential Containers Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir="$(dirname "$(readlink -f "$0")")"
+source "${script_dir}/lib.sh"
+
+export key_file="image_key"
+export key_path="/default/image_key/nginx"
+export trustee_repo="https://github.com/confidential-containers/trustee.git"
+export kbs_namespace="coco-tenant"
+export aa_kbc="${aa_kbc:-cc_kbc}"
+export kbs_svc_name="kbs"
+export kbs_ingress_name="kbs"
+export runtimeclass="kata-qemu"
+export encrypted_image="ghcr.io/confidential/nginx:latest"
+
+build_encryption_key() {
+	docker build -t unencrypted .
+	# Encryption key needs to be 32 byte sequence
+ 	head -c 32 /dev/urandom | openssl enc > "${key_file}"
+	# Encryption key needs to be provided as base 64 enconded string
+	key_b64="$(base64 < ${key_file})"
+	key_id="kbs://${key_path}"
+
+	# Image encryption logic
+	guest_components_repo="https://github.com/confidential-containers/guest-components.git"
+	guest_components_path="${script_dir}/guest-components"
+	if [ -d "${guest_components_path}" ]; then
+		echo "Guest components repo directory exists"
+	else
+		git clone "$guest_components_repo"
+	fi
+
+	pushd "${guest_components_path}"
+  		tag_name="coco-keyprovider"
+ 		docker build -t "${tag_name}" -f ./attestation-agent/docker/Dockerfile.keyprovider .
+   		mkdir -p oci/{input,output}
+   		skopeo copy docker-daemon:unencrypted:latest dir:./oci/input
```

Comment:
Once we get the basic flow working, let's think about adding a random secret to the image (rather than "something confidential") and checking for that value inside the guest. Also, we can grep for that string in the encrypted layer and make sure it doesn't show up.